app/jobs/create_virtual_objects_job.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+##
+# job to create virtual objects
+class CreateVirtualObjectsJob < GenericJob
+  queue_as :default
+
+  ##
+  # A job that creates virtual objects
+  # @param [Integer] bulk_action_id GlobalID for a BulkAction object
+  # @param [Hash] params additional parameters that an Argo job may need
+  # @option params [String] :create_virtual_objects CSV string
+  def perform(bulk_action_id, params)
+    super
+
+    virtual_objects = CsvToJsonService.convert(csv_string: params[:create_virtual_objects])
+
+    with_bulk_action_log do |log|
+      log.puts("#{Time.current} Starting #{self.class} for BulkAction #{bulk_action_id}")
+
+      parent_druids = virtual_objects[:virtual_objects].map { |hash| hash[:parent_id] }
+
+      # NOTE: We use this instead of `update_druid_count` because virtual object
+      #       creation does not use the `pids` form field.
+      bulk_action.update(druid_count_total: parent_druids.length)
+
+      # rubocop:disable Metrics/LineLength
+      problematic_druids = find_and_authorize_druids(parent_druids)
+      if problematic_druids[:unauthorized].any? || problematic_druids[:not_found].any?
+        log.puts("#{Time.current} Could not create virtual objects because user lacks ability to manage the following parent druids: #{problematic_druids[:unauthorized].to_sentence}") if problematic_druids[:unauthorized].any?
```

Comment:
I often find putting a bunch of the message into a var helps with long lines like this and makes the code more readable, IMO:

```
not_allowed_msg = "Could not create virtual objects because user lacks ability to manage the following parent druids"
log.puts("#{Time.current} #{not_allowed_msg}: #{problematic_druids[:unauthorized].to_sentence}") if problematic_druids[:unauthorized].any?
```

Reply:
Coming!